processor/processor.go

```diff
@@ -1033,6 +1041,10 @@ func (proc *HandleT) handlePendingGatewayJobs() bool {
 
 	proc.statDBR.End()
 
+	if enableDedup {
+		unprocessedList = proc.dedupHandler.DedupJobs(unprocessedList)
```

Comment:
We are writing to BadgerDB here. But, there is a chance that the application crashes before the processor marked the gateway jobs successful.

That will cause events to be dropped by the processor in the next loop. 

We should write to badger only after writing success to gateway jobsdb.